playground/frontend/playground_components/lib/src/models/example.dart

```diff
@@ -49,8 +53,11 @@ class Example extends ExampleBase {
     required this.logs,
     required this.outputs,
     required this.source,
+    ExampleLoadingDescriptor? descriptor,
     this.graph,
-  }) : super(
+  })  : descriptor =
+            descriptor ?? StandardExampleLoadingDescriptor(path: example.path),
```

Comment:
Second thoughts, it's better to not put the descriptor into the `Example` and just do `SnippetEditingController.setExample(example, descriptor: descriptor)`

This is better because from the domain POV the descriptor is not a part of the example. And this simplifies a lot of things.

Reply:
fixed